### ISSUE TITLE
Compile to ES5 and publish to NPM

### ISSUE THREAD

**Issue by awls99** [ospfranco/react-native-link-preview#21]
I'm trying to compile a project using this module, but upon running `npm run build` I get:
```> react-scripts build

Creating an optimized production build...
Failed to compile.

Failed to minify the code from this file: 

 	./node_modules/react-native-link-preview/index.js:7 

Read more here: http://bit.ly/2tRViJ9

```

from: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#npm-run-build-fails-to-minify 

```npm run build fails to minify
Some third-party packages don't compile their code to ES5 before publishing to npm. This often causes problems in the ecosystem because neither browsers (except for most modern versions) nor some tools currently support all ES6 features. We recommend to publish code on npm as ES5 at least for a few more years.


To resolve this:
Open an issue on the dependency's issue tracker and ask that the package be published pre-compiled.```

Please pre-compile and publish

**Comment by ospfranco** [comment]
weird... code should be ES5 compatible, can you copy and paste your `run` command?

**Comment by awls99** [comment]
I'm a bit new to this npm world, I [used create-react-app](https://github.com/facebook/create-react-app) to make kick off an react app which gave me this on the README:
```npm run build


Builds the app for production to the build folder.
It correctly bundles React in production mode and optimizes the build for the best performance.

The build is minified and the filenames include the hashes.
Your app is ready to be deployed!

See the section about deployment for more information.


```
which started to fail with the error after adding read-native-link-preview.
But yeah, I'm not sure where the build run commands pulls from

**Comment by ospfranco** [comment]
Ok, first thing: Seems to me what you are seeing is some webpack error, because it is trying to compile your node_modules folder, try to take a look into your webpack configuration, look into your package.json file and see what the `run` command is doing.

Second thing and most important: This package does not work on CORS protected environments, this means most browsers (Google chrome, Firefox, Edge, etc) so even if you get this to compile it won't work, please read the README, this package will work on node.js back-ends and babel compiled react-native apps.

**Comment by awls99** [comment]
for curiosity's sake, the build script is this one: https://github.com/facebook/create-react-app/blob/next/packages/react-scripts/scripts/build.js

I did read the README, but for some unknown reason, some of the sites I'm targeting have `Access-Control-Allow-Origin:*` thus this would work :)

**Comment by ospfranco** [comment]
That is the scafolding script, when you execute `npm run build` it is actually doing a different thing, try to take a look inside your `package.json` file, you will see a `scripts` entry, within it you should find the `run` and `build` commands, more than likely that is running a webpack build, once you found it you will probably find a webpack/gulp file in the root of your project, try to take a look inside, if you want you can try to make this work maybe by tweaking your babel/webpack/gulp configuration.

As I said the library should be ES5 compatible so I need a little more information on what is actually failing.

There is a security implication regarding CORS and preflight checks, even though some sites might not protect against it (which they should) and/or need to accept all incoming requests, this is not consistent AT ALL and then developers (especially beginners) think it is the library that it is failing, as such I must repeat over and over again, this library is NOT meant to work on cors protected environments, if you are lucky (like in your case) it might work with a few sites but if you try to use this as a generic solution it will eventually fail.

**Comment by ospfranco** [comment]
thanks nicolas, published under version 1.3.4
